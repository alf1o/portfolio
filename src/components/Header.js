import React from 'react';
import PropTypes from 'prop-types';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';

/**
  Render the main header.
  The menu icon is used to open the `SideMenu`.
*/
function Header({ onMenuClick }) {
  return (
    <AppBar
      position="static"
    >
      <Toolbar>
        <IconButton
          onClick={onMenuClick}
          aria-label="Menu"
        >
          <MenuIcon
            color="secondary"
          />
        </IconButton>
        <Typography
          variant="title"
        >
          Alfio Parisi
        </Typography>
      </Toolbar>
    </AppBar>
  );
}

Header.propTypes = {
  onMenuClick: PropTypes.func.isRequired
};

export default Header;
