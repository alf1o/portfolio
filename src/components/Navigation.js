import React from 'react';
import PropTypes from 'prop-types';
import Hidden from '@material-ui/core/Hidden';
import Drawer from '@material-ui/core/Drawer';
import SwipeableDrawer from '@material-ui/core/SwipeableDrawer';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import Divider from '@material-ui/core/Divider';
import { NavLink } from 'react-router-dom';
import { withStyles } from '@material-ui/core/styles';

const drawerWidth = 240;

const styles = theme => ({
  container: {
    zIndex: 1,
    overflow: 'hidden',
    position: 'relative',
    display: 'flex',
    width: drawerWidth,
    [theme.breakpoints.up('md')]: {
      width: '100%'
    }
  },
  toolbar: theme.mixins.toolbar,
  drawerPaper: {
    width: drawerWidth,
    [theme.breakpoints.up('md')]: {
      position: 'relative',
      width: '100vw'
    }
  },
  list: {
    [theme.breakpoints.up('md')]: {
      display: 'flex',
      justifyContent: 'space-around'
    }
  },
  links: {
    [theme.breakpoints.up('md')]: {
      marginLeft: 0,
      marginRight: 0,
      paddingLeft: 0,
      paddingRight: 0,
      textAlign: 'center'
    }
  }
});

const routes = [
  {
    name: 'Home',
    path: '/'
  },
  {
    name: 'Projects',
    path: '/projects/Arcade Game'
  },
  {
    name: 'Contacts',
    path: '/contacts'
  },
  {
    name: 'About',
    path: '/about'
  }
];

/**
  Render the side menu for navigation.
  Use a swipeable drawer for mobile and a permanent drawer otherwise.
*/
// TODO: styles for active link
function Navigation({ classes, onDrawerSwipe, isOpen }) {
  const drawer = (
    <List
      className={classes.list}
    >
      {routes.map(route => (
        <ListItem key={route.name}
          component={NavLink}
          to={route.path}
          className={classes.links}
        >
          <ListItemText primary={route.name} />
        </ListItem>
      ))}
    </List>
  );

  return (
    <div className={classes.container}>
      <Hidden
        mdUp={true}
      >
        <SwipeableDrawer
          variant="temporary"
          anchor="left"
          open={isOpen}
          onOpen={onDrawerSwipe}
          onClose={onDrawerSwipe}
          classes={{
            paper: classes.drawerPaper
          }}
          ModalProps={{
            keepMounted: true
          }}
        >
          <div className={classes.toolbar} />
          <Divider />
          {drawer}
        </SwipeableDrawer>
      </Hidden>
      <Hidden
        smDown={true}
        implementation="css"
      >
        <Drawer
          anchor="top"
          variant="permanent"
          open={true}
          classes={{
            paper: classes.drawerPaper
          }}
        >
          {drawer}
        </Drawer>
      </Hidden>
    </div>
  );
}

Navigation.propTypes = {
  classes: PropTypes.object.isRequired,
  onDrawerSwipe: PropTypes.func.isRequired,
  isOpen: PropTypes.bool.isRequired
};

export { Navigation };
export default withStyles(styles)(Navigation);
