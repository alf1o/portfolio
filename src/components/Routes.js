import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import Home from './Home';
import Contacts from './Contacts';
import About from './About';
import ProjectTabs from './ProjectTabs';
import Project from './Project';
import NotFound from './NotFound';
import { Switch, Route } from 'react-router-dom';

function Routes({ repos, text }) {
  return (
    <Switch>
      <Route
        exact={true}
        path="/"
        component={Home}
      />
      <Route
        path="/contacts"
        component={Contacts}
      />
      <Route
        path="/about"
        render={match => (
          <About text={text} />
        )}
      />
      <Route
        path="/projects"
        render={match => (
          <Fragment>
            <ProjectTabs
              match={match}
              projects={repos}
            />
            {repos.map(repo => (
              <Route key={repo.name}
                path={`/projects/${repo.name}`}
                render={match => (
                  <Project
                    project={repo}
                  />
                )}
              />
            ))}
          </Fragment>
        )}
      />
      <Route
        component={NotFound}
      />
    </Switch>
  );
}

Routes.propTypes = {
  repos: PropTypes.array.isRequired,
  text: PropTypes.string.isRequired
};

export default Routes;
