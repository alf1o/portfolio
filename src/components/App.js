import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import Header from './Header';
import ProjectTabs from './ProjectTabs';
import withRoot from '../withRoot';

class App extends Component {

  static propTypes = {
    repos: PropTypes.array.isRequired
  };

  render() {
    const { repos } = this.props;
    return (
      <Fragment>
        <Header />
        <ProjectTabs projects={repos} />
      </Fragment>
    );
  }
}

export { App };
export default withRoot(App);
