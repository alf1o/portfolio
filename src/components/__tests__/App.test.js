import React from 'react';
import { App } from '../App';
import { createShallow } from '@material-ui/core/test-utils';
import Header from '../Header';
import Navigation from '../Navigation';
import Routes from '../Routes';

describe('`App`', () => {
  let props = {};
  let mountedApp;
  const shallow = createShallow();

  function app() {
    if (!mountedApp) mountedApp = shallow(<App {...props} />);
    return mountedApp;
  }

  beforeEach(() => {
    props = {
      classes: {
        container: '',
        routesContainer: ''
      },
      repos: [],
      text: 'Test text'
    };
    mountedApp = undefined;
  });

  it('should have a `state.mobileOpen` property', () => {
    expect(app().state().mobileOpen).toBeDefined();
  });

  it('should have an `handleDrawerToggle` method', () => {
    expect(App.prototype.handleDrawerToggle).toBeDefined();
  });

  it('should pass `handleDrawerToggle` to `Header`', () => {
    const header = app().find(Header);
    expect(header.props().onMenuClick).toEqual(app().instance().handleDrawerToggle);
  });

  it('should pass `handleDrawerToggle` to `Navigation`', () => {
    const nav = app().find(Navigation);
    expect(nav.props().onDrawerSwipe).toEqual(app().instance().handleDrawerToggle);
  });

  it('should pass `mobileOpen` to `Navigation`', () => {
    const nav = app().find(Navigation);
    expect(nav.props().isOpen).toBe(app().state().mobileOpen);
  });

  it('should pass `repos` to `Routes`', () => {
    const rts = app().find(Routes);
    expect(rts.prop('repos')).toEqual(app().instance().props.repos);
  });

  it('should pass `repos` to `Routes`', () => {
    const rts = app().find(Routes);
    expect(rts.prop('text')).toBe(app().instance().props.text);
  });

  describe('`handleDrawerToggle`', () => {
    const hDTSpy = jest.spyOn(App.prototype, 'handleDrawerToggle');
    afterEach(() => {
      hDTSpy.mockClear();
    });

    it('should toggle `mobileOpen`', () => {
      expect(app().state().mobileOpen).toBe(false);
      hDTSpy.call(app().instance());
      expect(app().state().mobileOpen).toBe(true);
      hDTSpy.call(app().instance());
      expect(app().state().mobileOpen).toBe(false);
    });
  });

  it('shoud match snapshot', () => {
    expect(app()).toMatchSnapshot();
  });
});
