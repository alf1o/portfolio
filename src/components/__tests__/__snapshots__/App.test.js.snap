// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`App\` shoud match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    classes={
        Object {
            "container": "",
            "routesContainer": "",
          }
    }
    repos={Array []}
    text="Test text"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Header
          onMenuClick={[Function]}
/>,
        <WithStyles(Paper)
          className=""
>
          <WithStyles(Navigation)
                    isOpen={false}
                    onDrawerSwipe={[Function]}
          />
          <WithStyles(Paper)
                    className=""
                    elevation={0}
          >
                    <Routes
                              repos={Array []}
                              text="Test text"
                    />
          </WithStyles(Paper)>
</WithStyles(Paper)>,
        <ButtonsContainer />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onMenuClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(Navigation)
              isOpen={false}
              onDrawerSwipe={[Function]}
/>,
            <WithStyles(Paper)
              className=""
              elevation={0}
>
              <Routes
                            repos={Array []}
                            text="Test text"
              />
</WithStyles(Paper)>,
          ],
          "className": "",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "isOpen": false,
              "onDrawerSwipe": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Routes
                repos={Array []}
                text="Test text"
/>,
              "className": "",
              "elevation": 0,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "repos": Array [],
                "text": "Test text",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Header
            onMenuClick={[Function]}
/>,
          <WithStyles(Paper)
            className=""
>
            <WithStyles(Navigation)
                        isOpen={false}
                        onDrawerSwipe={[Function]}
            />
            <WithStyles(Paper)
                        className=""
                        elevation={0}
            >
                        <Routes
                                    repos={Array []}
                                    text="Test text"
                        />
            </WithStyles(Paper)>
</WithStyles(Paper)>,
          <ButtonsContainer />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onMenuClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(Navigation)
                isOpen={false}
                onDrawerSwipe={[Function]}
/>,
              <WithStyles(Paper)
                className=""
                elevation={0}
>
                <Routes
                                repos={Array []}
                                text="Test text"
                />
</WithStyles(Paper)>,
            ],
            "className": "",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "isOpen": false,
                "onDrawerSwipe": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Routes
                  repos={Array []}
                  text="Test text"
/>,
                "className": "",
                "elevation": 0,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "repos": Array [],
                  "text": "Test text",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {},
  },
}
`;
